### PR TITLE
 Update bundle build (#112)

### PR DESCRIPTION
**Pull request by karelmaxa**
Packaging of all modules (except UI modules) has been switched to the jar packaging. 

Usage of some deprecated stuff has been removed.

### CODE REVIEW

File: openidm-ui/openidm-ui-enduser/pom.xml [hunk 1]
```diff
@@ -1,164 +1,135 @@
...
+            <groupId>org.forgerock.openidm</groupId>
+            <artifactId>openidm-ui-common</artifactId>
+            <version>${project.version}</version>
+            <type>zip</type>
+            <classifier>www</classifier>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.servicemix.bundles</groupId>
+            <artifactId>org.apache.servicemix.bundles.rhino</artifactId>
```
Comment (pavelhoral) [inline openidm-ui/openidm-ui-enduser/pom.xml:57]: Is Rhino required? I know that it was there before the refactor, but I am wondering what it is being used for in this bundle.

File: pom.xml [hunk 2]
```diff
@@ -187,6 +198,23 @@
...
         <apache.commons-fileupload.version>1.4</apache.commons-fileupload.version>
+
+        <!-- ************** -->
+        <!-- Build settings -->
+        <!-- ************** -->
+
+        <!-- i18n-maven-plugin -->
+        <i18nPluginVersion>1.4.1</i18nPluginVersion>
+        <!-- maven-jar-plugin http://maven.apache.org/plugins/maven-jar-plugin/ -->
+        <mavenJarPluginVersion>3.2.0</mavenJarPluginVersion>
```
Comment (pavelhoral) [inline pom.xml:209]: Why overriding wrensec-parent?

Comment (pavelhoral) [reply pom.xml:209]: Same for plugins below this one.

File: pom.xml [hunk 3]
```diff
@@ -230,51 +258,150 @@
...
-                                    <goal>generate</goal>
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
+                <!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on the Maven build itself. -->
+                <plugin>
+                    <groupId>org.eclipse.m2e</groupId>
+                    <artifactId>lifecycle-mapping</artifactId>
+                    <version>1.0.0</version>
```
Comment (pavelhoral) [inline pom.xml:303]: I am all for replacing this plugin with processing instructions (if possible):

* https://www.eclipse.org/m2e/documentation/release-notes-17.html#new-syntax-for-specifying-lifecycle-mapping-metadata
* https://github.com/WrenSecurity/wrensec-parent/blob/master/pom.xml#L1026

**Review by pavelhoral — CHANGES_REQUESTED** [review]
Looks good, just a few comments.

File: openidm-provisioner/pom.xml [hunk 4]
```diff
@@ -100,21 +91,14 @@
     </dependencies>
 
     <build>
-        <finalName>${project.artifactId}</finalName>
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply openidm-provisioner/pom.xml:103]: I thought that this configuration option is inherited from the parent pom. But it is not.

File: openidm-provisioner-openicf/pom.xml [hunk 5]
```diff
@@ -174,21 +163,16 @@
     </dependencies>
 
     <build>
-        <finalName>${project.artifactId}</finalName>
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply openidm-provisioner-openicf/pom.xml:177]: I thought that this configuration option is inherited from the parent pom. But it is not.

File: openidm-router/pom.xml [hunk 6]
```diff
@@ -105,43 +102,30 @@
...
-                        <include>**</include>
-                    </includes>
-                </configuration>
-
                 <executions>
                     <execution>
-                        <goals>
-                            <goal>test-jar</goal>
-                        </goals>
+                    <goals>
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply openidm-router/pom.xml:126]: Fixed.

File: openidm-ui/openidm-ui-api/pom.xml [hunk 7]
```diff
@@ -1,34 +1,39 @@
...
-    <description>
-        A Swagger-based tool for interacting with most of the endpoints provided with a default IDM installation.
-    </description>
     <packaging>bundle</packaging>
 
+    <name>Wren:IDM - UI - API Explorer</name>
+    <description>A Swagger-based tool for interacting with most of the endpoints provided with a default IDM installation.</description>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply openidm-ui/openidm-ui-api/pom.xml:34]: The same configuration is in the parent pom, removing.

File: openidm-workflow-activiti/src/main/java/org/forgerock/openidm/workflow/activiti/impl/TaskInstanceHistoryResource.java [hunk 8]
```diff
@@ -297,7 +297,7 @@ private void setSortKeys(HistoricTaskInstanceQuery query, QueryRequest request)
                         query.orderByHistoricTaskInstanceDuration();
                         break;
                     case ActivitiConstants.ACTIVITI_STARTTIME:
-                        query.orderByHistoricTaskInstanceStartTime();
+                        query.orderByTaskCreateTime();
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply openidm-workflow-activiti/src/main/java/org/forgerock/openidm/workflow/activiti/impl/TaskInstanceHistoryResource.java:300]: It should be. Citation from Javadoc: `@deprecated Use {@link #orderByTaskCreateTime()}`.

File: openidm-ui/openidm-ui-enduser/pom.xml (hunk 1 above)
Comment (karelmaxa) [reply openidm-ui/openidm-ui-enduser/pom.xml:65]: I'm not sure. I can create dedicated issue to check it.

File: pom.xml (hunk 2 above)
Comment (karelmaxa) [reply pom.xml:209]: I have missed the definitions in the parent pom, removed.

File: pom.xml (hunk 3 above)
Comment (karelmaxa) [reply pom.xml]: Plugin configuration has been completely removed.

File: openidm-ui/openidm-ui-admin/pom.xml [hunk 9]
```diff
@@ -129,23 +120,10 @@
                 </executions>
             </plugin>
 
-
             <plugin>
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
                 <extensions>true</extensions>
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply openidm-ui/openidm-ui-admin/pom.xml:130]: Bundle plugin runs twice, but the manifest is generated just in the first run. It should be acceptable.

**Pull request merged**